src/global_test/mod.rs

```diff
@@ -0,0 +1,239 @@
+//! # Test module
+//!
+//! This module contain all the initialization and global functions needed by the tests
+#[cfg(test)]
+pub mod tests {
+    use crate::{model, regiment};
+
+    /// # Initialize the stats structs with some values given as parameters
+    ///
+    /// # Parameters
+    /// advance (usize): the advance statistic of the model
+    ///
+    /// march (usize): the march statistic of the model
+    ///
+    /// discipline (usize): the discipline statistic of the model
+    ///
+    /// health_point (usize): the health_point statistic of the model
+    ///
+    /// defense (usize): the defense statistic of the model
+    ///
+    /// resilience (usize): the resilience statistic of the model
+    ///
+    /// armour (usize): the armour statistic of the model
+    ///
+    /// aegis (usize): the aegis statistic of the model
+    ///
+    /// attack (usize): the attack statistic of the model
+    ///
+    /// offensive (usize): the offensive statistic of the model
+    ///
+    /// strength (usize): the strength statistic of the model
+    ///
+    /// armour_penetration (usize): the armour_penetration statistic of the model
+    ///
+    /// agility (usize): the agility statistic of the model
+    /// # Return
+    /// Stats Model: The Stats Model created and initialized
```

Comment:
/!\ "The Stats Model", I'm not sure that I get this, maybe go for something like this
The Stats created and initialized